dali/imgcodec/image_format.cc

```diff
@@ -67,5 +67,19 @@ span<ImageFormat* const> ImageFormatRegistry::Formats() const {
   return make_cspan(format_ptrs_);
 }
 
+size_t ImageParser::ReadHeader(ImageSource *encoded, uint8_t *buffer, size_t n) const {
+  if (encoded->Kind() == InputKind::HostMemory) {
+    if (encoded->Size() < n)
+      n = encoded->Size();
+    std::copy_n(encoded->RawData<char>(), n, buffer);
+  } else {
+    auto stream = encoded->Open();
```

Comment:
perhaps for now we could assert that it's not DeviceMem ?

Reply:
Why? Know it will just fail to open ant that's OK. We may add a reader for device memory in the future.

Reply:
I didn't notice there's a failure in Open when I wrote this. Nevermind